### PR TITLE
FIX #549: Always on top window flag.

### LINKED ISSUE #549: Modal electron window not working
### Description

We should be able to specify a nativefier flag to indicate that we want to maintain the nativefied app "always on top"

### Steps to reproduce issue
Just nativefy any public website with any flag combination and the result will be always a non modal window app.

I've tried to add the flag manually on main.js on all  new _electron.BrowserWindow({ ... modal:true}) but the created windows seems to ignore this flag.

### Details
- I've tried with a bunch of public websites and the result is always the same a non modal windows.
- Version of Nativefier: `v7.5.4`
- Version of node.js: `v8.2.1`
- OS: `linux`
- Error message / stack trace (if any): No errors since this can be an inprovement.

### PR DESCRIPTION
**Pull request by sgroh**
Modal feature added using --always-on-top build-time flag.

### CODE REVIEW

**Comment by ronjouch** [comment]
@sgroh you must have forgotten to push files, it doesn't look like this code does anything.

**Comment by sgroh** [comment]
@ronjouch what do you mean? The only thing you need to do is propagate the flag `alwayOnTop` to the options that  `BrowserWindow` need in it's creation. Please, have a look to `mainWindow.js`

The resulting code let the nativefier user create an application on a modal window only if the flag is specified, otherwise we had the regular behavior we had before introduce this change.

**Comment by sgroh** [comment]
@ronjouch  you can also check https://github.com/electron/electron/blob/master/docs/api/browser-window.md where is documented the `alwayOnTop` option attribute. Please let me know if you still think that there is something missing on this PR. Thanks.

**Comment by ronjouch** [comment]
> "*you can also check https://github.com/electron/electron/blob/master/docs/api/browser-window.md where is documented the `alwayOnTop` option attribute. Please let me know if you still think that there is something missing on this PR. Thanks.*"

@sgroh thanks for precising this. I was asking because, from your comment https://github.com/jiahaog/nativefier/issues/549#issuecomment-370585859 I had understood you planned to rely on some `electron-modal` third-party module, so it seemed fishy to not see it as new dependency. Well, glad it's out-of-the-box Electron 👍. Will try your PR.

File: app/src/components/mainWindow/mainWindow.js [hunk 1]
```diff
@@ -89,6 +89,8 @@ function createMainWindow(inpOptions, onAppQuit, setDockBadge) {
     icon: getAppIcon(),
     // set to undefined and not false because explicitly setting to false will disable full screen
     fullscreen: options.fullScreen || undefined,
+    // window modal defaulted to false
```
Comment (ronjouch) [inline app/src/components/mainWindow/mainWindow.js]: Unnecessary comment, and "modal" means something different than "always on top", pls remove

Comment (sgroh) [reply app/src/components/mainWindow/mainWindow.js]: sure yes.

Comment (sgroh) [reply app/src/components/mainWindow/mainWindow.js]: PR updated! Thanks.

**Comment by sgroh** [comment]
@ronjouch is the PR looking good now?

**Comment by ronjouch** [comment]
@sgroh doesn't work for me under Linux. Under what OS does it work for you? [Googling a bit](https://www.google.com/search?q=electron alwaysOnTop linux), I see other Electron users complaining the `alwaysOnTop` flag is buggy. Amending your documentation with the mention that it's flakey and redirecting bugs to Electron.

**Pull request merged**